WordPress/src/main/java/org/wordpress/android/ui/mysite/ImprovedMySiteFragment.kt

```diff
@@ -1,9 +1,78 @@
 package org.wordpress.android.ui.mysite
 
+import android.os.Bundle
+import android.os.Parcelable
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
 import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.ViewModelProviders
+import androidx.recyclerview.widget.LinearLayoutManager
+import kotlinx.android.synthetic.main.new_my_site_fragment.*
+import org.wordpress.android.R
+import org.wordpress.android.WordPress
+import javax.inject.Inject
 
 class ImprovedMySiteFragment : Fragment() {
+    @Inject lateinit var viewModelFactory: ViewModelProvider.Factory
+    private lateinit var viewModel: MySiteViewModel
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        (requireActivity().application as WordPress).component().inject(this)
+        viewModel = ViewModelProviders.of(this, viewModelFactory).get(MySiteViewModel::class.java)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(
+                R.layout.new_my_site_fragment,
+                container,
+                false
+        )
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val layoutManager = LinearLayoutManager(activity)
+
+        savedInstanceState?.getParcelable<Parcelable>(KEY_LIST_STATE)?.let {
+            layoutManager.onRestoreInstanceState(it)
+        }
+
+        recycler_view.layoutManager = layoutManager
+
+        viewModel.uiModel.observe(viewLifecycleOwner, {
+            it?.let { items ->
+                loadData(items)
+            }
+        })
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        recycler_view.layoutManager?.let {
+            outState.putParcelable(KEY_LIST_STATE, it.onSaveInstanceState())
+        }
+    }
+
+    private fun loadData(items: List<MySiteItem>) {
+        if (recycler_view.adapter == null) {
+            recycler_view.adapter = MySiteAdapter()
+        }
+        val adapter = recycler_view.adapter as MySiteAdapter
+        val recyclerViewState = recycler_view.layoutManager?.onSaveInstanceState()
+        adapter.loadData(items)
+        recycler_view.layoutManager?.onRestoreInstanceState(recyclerViewState)
```

Comment:
I'm not sure I understand why we need to call `onSaveInstanceState()` and `onRestoreInstanceState()` here.

Reply:
hmm, good question. Generally it's there to keep scroll position after data changes but I think in this case it's not necessary (cause we don't expect massive changes to the data). I'll remove it.